### ISSUE TITLE
FP needless-raw-string-hashes

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#11068]
### Summary

.

### Lint Name

needless-raw-string-hashes

### Reproducer

I tried this code:

```rust
#![allow(clippy::eq_op)]

pub fn main() {
    assert_eq!(r##"a"#"b"##, "a\"#\"b");
}

```

I saw this happen:

```
warning: unnecessary hashes around raw string literal
 --> ./tests/ui/raw-str.rs:4:16
  |
4 |     assert_eq!(r##"a"#"b"##, "a\"#\"b");
  |                ^^^^^^^^^^^^ help: try: `r#"a"#"b"#`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_raw_string_hashes
  = note: `#[warn(clippy::needless_raw_string_hashes)]` on by default
```

I expected to see this happen:

suggested code.
````
#![allow(clippy::eq_op)]

pub fn main() {
    assert_eq!(r#"a"#"b"#, "a\"#\"b");
}

````
=>

```
error: no rules expected the token `"b"`
  --> ./tests/ui/raw-str.rs:4:22
   |
4  |     assert_eq!(r#"a"#"b"#, "a\"#\"b");
   |                      ^^^ no rules expected this token in macro call
   |
note: while trying to match `,`
  --> /home/matthias/.rustup/toolchains/master/lib/rustlib/src/rust/library/core/src/macros/mod.rs:37:16
   |
37 |     ($left:expr, $right:expr $(,)?) => {
   |                ^

error: aborting due to previous error
```

### Version

```text
rustc 1.72.0-nightly (32d81eccd 2023-07-02)
binary: rustc
commit-hash: 32d81eccd64513bacef9dfa1574543ada6b45d85
commit-date: 2023-07-02
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.5
```

**Comment by Centri3** [comment]
Ah, it doesn't check if the `"` is not following a quote :sweat_smile: so it resets the number needed

@rustbot claim
